.circleci/config.yml

```diff
@@ -27,7 +46,23 @@ jobs:
       - restore_cache: { key: 'gradle-wrapper-v2-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}' }
       - restore_cache: { key: 'gradle-cache-v2-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}' }
       - restore_cache: { key: 'graal-cache' }
-      - run: sudo apt-get update && sudo apt-get install build-essential zlib1g-dev
+      - run: (sudo apt-get update || sudo apt-get update) && sudo apt-get install build-essential zlib1g-dev
+      - run: ./gradlew --profile --parallel --stacktrace --continue check
+      - save_cache:
+          key: 'graal-cache'
+          paths: [ ~/.gradle/caches/com.palantir.graal ]
+      - store_test_results: { path: build/test-results }
+      - store_artifacts: { path: build/reports/tests }
+
+  check-jdk11:
+    docker: [{ image: 'circleci/openjdk:11-node' }]
+    resource_class: medium+
+    steps:
+      - attach_workspace: { at: . }
+      - restore_cache: { key: 'gradle-wrapper-v2-jdk11-{{ checksum "gradle/wrapper/gradle-wrapper.properties" }}' }
+      - restore_cache: { key: 'gradle-cache-v2-jdk11-{{ checksum "versions.props" }}-{{ checksum "build.gradle" }}' }
+      - restore_cache: { key: 'graal-cache' }
+      - run: (sudo apt-get update || sudo apt-get update) && sudo apt-get install build-essential zlib1g-dev
       - run: ./gradlew --profile --parallel --stacktrace --continue check
       - save_cache:
           key: 'graal-cache'
```

Comment:
also we don't need to be saving this cache twice (we're already doing it in `check`).
Not to mention I don't believe this never changes, so we should probably checksum some files into this cache key...